### PR TITLE
Ros2 main update

### PR DESCRIPTION
**Pull request by fmrico**
Dear all,

This PR contains:
* Adding `pluginlib` dependency in `CMakeLists.txt` and `package.xml`.
* Fixes for deprecation in `TYPED_TEST_CASE`, `rosbag` API, and `rclcppp::Duration` constructor.

All these deprecations are with respect to ROS2 main branch.

I hope it helps!

### CODE REVIEW

**Comment by SteveMacenski** [comment]
CI build failures https://app.circleci.com/pipelines/github/ANYbotics/grid_map/193/workflows/095b86a1-1c8c-4826-9a02-15906fd79e81/jobs/200

File: grid_map_cv/CMakeLists.txt [hunk 1]
```diff
@@ -29,6 +30,7 @@ set(dependencies
   sensor_msgs
   rclcpp
   OpenCV
+  pluginlib
```
Comment (maximilianwulf) [inline grid_map_cv/CMakeLists.txt:33]: Hey thank you for the MR. Pluginlib should be an indirect dependency through the filters package, at least for ROS1. https://github.com/ros/filters/blob/lunar-devel/package.xml

Did you experience compilation issues?

**Review by maximilianwulf — COMMENTED** [review]
Thank you very much for the MR submission.

**Comment by SteveMacenski** [comment]
#296 supersedes

**Pull request closed without merging**